data-protocols/ids/ids-api-multipart-dispatcher-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/client/sender/MultipartArtifactRequestSender.java

```diff
@@ -0,0 +1,102 @@
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Fraunhofer Institute for Software and Systems Engineering - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.ids.api.multipart.client.sender;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.fraunhofer.iais.eis.ArtifactRequestMessageBuilder;
+import de.fraunhofer.iais.eis.DynamicAttributeToken;
+import de.fraunhofer.iais.eis.Message;
+import de.fraunhofer.iais.eis.util.Util;
+import okhttp3.OkHttpClient;
+import org.eclipse.dataspaceconnector.ids.api.multipart.client.message.IdsMultipartParts;
+import org.eclipse.dataspaceconnector.ids.api.multipart.client.message.MultipartRequestInProcessResponse;
+import org.eclipse.dataspaceconnector.ids.spi.IdsId;
+import org.eclipse.dataspaceconnector.ids.spi.IdsType;
+import org.eclipse.dataspaceconnector.ids.spi.transform.TransformerRegistry;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.iam.IdentityService;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataRequest;
+
+import java.net.URI;
+
+public class MultipartArtifactRequestSender extends IdsMultipartSender<DataRequest, MultipartRequestInProcessResponse> {
+
+    private final TransformerRegistry transformerRegistry;
+
+    public MultipartArtifactRequestSender(String connectorId,
+                                          OkHttpClient httpClient,
+                                          ObjectMapper objectMapper,
+                                          Monitor monitor,
+                                          IdentityService identityService,
+                                          TransformerRegistry transformerRegistry) {
+        super(connectorId, httpClient, objectMapper, monitor, identityService);
+        this.transformerRegistry = transformerRegistry;
+    }
+
+    @Override
+    public Class<DataRequest> messageType() {
+        return DataRequest.class;
+    }
+
+    @Override
+    protected String getConnectorId(DataRequest request) {
+        return request.getConnectorId();
+    }
+
+    @Override
+    protected String getConnectorAddress(DataRequest request) {
+        return request.getConnectorAddress();
+    }
+
+    @Override
+    protected Message buildMessageHeader(DataRequest request, DynamicAttributeToken token) throws Exception {
+        var asset = request.getAsset();
+        IdsId id = IdsId.Builder.newInstance()
+                .value(asset.getId())
+                .type(IdsType.ARTIFACT)
+                .build();
+        var transformationResult = transformerRegistry.transform(id, URI.class);
+        if (transformationResult.hasProblems()) {
+            throw new EdcException("Failed to create artifact ID from asset.");
```

Comment:
Not to change now (can be done in a later PR), but can we report the errors and return an error message to the client?